package/MDAnalysis/analysis/rms.py

```diff
@@ -189,6 +190,7 @@ def rmsd(a, b, weights=None, center=False, superposition=False):
             raise ValueError('weights must have same length as a/b')
         # weights are constructed as relative to the mean
         relative_weights = np.asarray(weights) / np.mean(weights)
+        relative_weights = relative_weights.astype(np.float64)
```

Comment:
for what its worth you can do this in one go, `relative_weights = ((np.asarray(weights) / np.mean(weights)).astype(np.float64))`